packages/edit-site/src/components/global-styles/font-library-modal/style.scss

```diff
@@ -35,9 +34,19 @@ $footer-height: 70px;
 }
 
 .font-library-modal__tabpanel-layout {
-
 	margin-top: $grid-unit-40;
 
+	.font-library-modal__loading {
+		width: 100%;
+		height: 100%;
+		display: flex;
+		position: absolute;
+		left: 0;
+		top: $header-height;
```

Comment:
```suggestion
		top: 0;
```

It looks like the current loading area doesn't match the size of the modal content. I also think the `$header-height` CSS variable is inappropriate here since it represents the header size of the editor itself.

I think if we just change it to `top:0`, the loading area and modal content will match in size and the `ProgressBar` will be centered exactly.

### Before

![image](https://github.com/WordPress/gutenberg/assets/54422211/e09f6c62-0e2f-4c3b-bd3a-bf475c9749db)

### After

![image](https://github.com/WordPress/gutenberg/assets/54422211/bf6df3f8-7488-4e4d-9f18-ea8bb9827b72)

Reply:
I used `$header-height` because this is the variable that's used to set the height of the modal header [here](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/guide/style.scss#L23).

I think we need to offset the centering position, otherwise the ProgressBar isn't centered within the content area (the area below the header and tabs, and the bottom of the modal):

| Top | Bottom |
| ----- | ------ |
| ![image](https://github.com/WordPress/gutenberg/assets/1645628/4f6cef41-e37a-478d-b8a5-f96322d66955) | ![image](https://github.com/WordPress/gutenberg/assets/1645628/d7240762-da77-4db2-9228-543638374f15) |

Offset by 60px / header-height:

| Top | Bottom |
| ----- | ------ |
| ![image](https://github.com/WordPress/gutenberg/assets/1645628/a03f6e43-5341-4977-aad2-ce5215cddb9a) | ![image](https://github.com/WordPress/gutenberg/assets/1645628/09bec2ee-6653-43a9-8c5a-b92fc606f599) |

But I don't think I like using the `top` position for this, maybe it would be better as a `padding-top` setting?

![image](https://github.com/WordPress/gutenberg/assets/1645628/88fe62d3-4fed-488f-843e-1f37d46edaa2)

I've gone ahead and changed this for now in: https://github.com/WordPress/gutenberg/pull/60570/commits/7dafc44afab986ff934e47a4589e4eb53cec55fd.

Reply:
> I used `$header-height` because this is the variable that's used to set the height of the modal header [here](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/guide/style.scss#L23).

Oh, I forgot that this CSS variable was also used in the modal. However, I believe this is the height of the `Guide` component's header. I think the height of the `Modal` component' header is below.

https://github.com/WordPress/gutenberg/blob/68903c809f5db60885964747f49b2a285577dafd/packages/components/src/modal/style.scss#L101

> But I don't think I like using the `top` position for this, maybe it would be better as a `padding-top` setting?

I think we can adjust either the `top` or `padding-top`, but we need to use a value that takes into account both the height of the modal header and the height of the tabs. In other words, it is the sum of the modal height (`$header-height + $grid-unit-15`) and the tab height (`$grid-unit-60`). It should be `$header-height + $grid-unit-15 + $grid-unit-60` ( 60px + 12px + 48px = **120px**).

Below is a screenshot when the value of `top` remains zero and `padding-top: $header-height + $grid-unit-15 + $grid-unit-60` is applied.

![modal-padding-top](https://github.com/WordPress/gutenberg/assets/54422211/a0ff418f-9926-4438-b3f2-a6bac13c62eb)

Reply:
I added the top padding by 89e2314. Please let me know if this works correctly in your environment.

Reply:
We don't need to take into account the height of the tabs ($grid-unit-60), as they're set to position: sticky and therefore don't affect the centering calculation. You can see in your above screenshot that this pushes the ProgressBar off-center as it has too much spacing above it.

I did start with the above calculation you stated but noticed that the position of the ProgressBar is incorrect. We only need to account for 60px, which gives us these equal top and bottom measurements:

| Top | Bottom |
| ----- | ------ |
| ![image](https://github.com/WordPress/gutenberg/assets/1645628/a03f6e43-5341-4977-aad2-ce5215cddb9a) | ![image](https://github.com/WordPress/gutenberg/assets/1645628/09bec2ee-6653-43a9-8c5a-b92fc606f599) |

Reply:
I see, I would like to check it out too, so please push that change 👍

Reply:
Hmm, today when I test this, the latest commit using `$header-height + $grid-unit-15 + $grid-unit-60` also looks centered. I'm a little confused as this is not what I was seeing before when I first tested using these variables, but I'm happy to go with it if it's working now 😅 I can see in my debug tools that it's coming through as 120px, and it looks correct to me 👍

To confirm, the screenshots in [this comment](https://github.com/WordPress/gutenberg/pull/60570#discussion_r1563822817) were using the 60px offset value rather than 120px, but also confirming that I'm seeing a nicely centered ProgressBar with 120px ✨

I'm happy to stick with `$header-height + $grid-unit-15 + $grid-unit-60` as this makes the most logical sense and also explains itself very nicely, i.e. it's clear why this spacing is needed.

Reply:
I'm glad that you seem to have gotten the same results. This PR has already been approved and the code looks good to me, so please feel free to merge 👍

Reply:
Thanks so much for your help and reviews here, @t-hamano! I'll go ahead and merge.